flink/src/main/java/org/apache/iceberg/flink/FlinkFilters.java

```diff
@@ -0,0 +1,183 @@
+
+        case IN:
+          List<ResolvedExpression> args = call.getResolvedChildren();
+          FieldReferenceExpression field = (FieldReferenceExpression) args.get(0);
+          List<ResolvedExpression> values = args.subList(1, args.size());
+
+          List<Object> inputValues = values.stream().filter(
+              expression -> {
+                if (expression instanceof ValueLiteralExpression) {
+                  return !((ValueLiteralExpression) flinkExpression).isNull();
+                }
+
+                return false;
+              }
+          ).map(expression -> {
+            ValueLiteralExpression valueLiteralExpression = (ValueLiteralExpression) expression;
+            return valueLiteralExpression.getValueAs(valueLiteralExpression.getOutputDataType().getConversionClass())
+                .get();
+          }).collect(Collectors.toList());
+          return Optional.of(Expressions.in(field.getName(), inputValues));
+
+        case NOT:
+          Optional<Expression> child = convert(call.getResolvedChildren().get(0));
+          if (child.isPresent()) {
+            return Optional.of(Expressions.not(child.get()));
+          }
+
+          return Optional.empty();
+
+        case AND: {
+          return convertLogicExpression(Expressions::and, call);
+        }
+
+        case OR: {
+          return convertLogicExpression(Expressions::or, call);
+        }
+      }
+    }
+
+    return Optional.empty();
+  }
+
+  private static Optional<Expression> convertLogicExpression(BiFunction<Expression, Expression, Expression> function,
+                                                             CallExpression call) {
+    List<ResolvedExpression> args = call.getResolvedChildren();
+    Optional<Expression> left = convert(args.get(0));
+    Optional<Expression> right = convert(args.get(1));
+    if (left.isPresent() && right.isPresent()) {
+      return Optional.of(function.apply(left.get(), right.get()));
+    }
+
+    return Optional.empty();
+  }
+
+  private static Optional<Expression> convertComparisonExpression(BiFunction<String, Object, Expression> function,
+                                                                  CallExpression call) {
+    List<ResolvedExpression> args = call.getResolvedChildren();
+    FieldReferenceExpression fieldReferenceExpression;
+    ValueLiteralExpression valueLiteralExpression;
+    if (literalOnRight(args)) {
+      fieldReferenceExpression = (FieldReferenceExpression) args.get(0);
+      valueLiteralExpression = (ValueLiteralExpression) args.get(1);
+    } else {
+      fieldReferenceExpression = (FieldReferenceExpression) args.get(1);
+      valueLiteralExpression = (ValueLiteralExpression) args.get(0);
+    }
+
+    String name = fieldReferenceExpression.getName();
+    Class clazz = valueLiteralExpression.getOutputDataType().getConversionClass();
+    Object value = valueLiteralExpression.getValueAs(clazz).get();
```

Comment:
`ValueLiteralExpression` allows the value to be null, in which case `get` here will throw an exception. How is this avoided? Does the parser reject `col = null` expressions?

@openinx may be able to help here, too.

Reply:
Actually, a few lines down there is an assertion that the value isn't null. This looks like a bug to me.

Reply:
I tested it ,if the sql is `select * from mytable where data = null` ,it do not supports filter push down in flink,and we do not can get any data.
if the sql is `select * from mytable where data is null`, it is normal ,and It will enter the `IS_NULL` branch of switch

Reply:
I think it would be good to fix this case rather than making the assumption that Flink won't push the `= null` filter. Handling null will be good for maintainability.

Reply:
I agreed with @rdblue  that handling null in this function rather than assuming flink won't push down the `null`.